lib/elixir/lib/string.ex

```diff
@@ -2502,18 +2563,22 @@ defmodule String do
   end
 
   defp string_to_bag(string, bag, length) do
-    case next_grapheme(string) do
-      {char, rest} ->
+    case :unicode_util.gc(string) do
+      [gc | rest] ->
         bag =
           case bag do
-            %{^char => current} -> %{bag | char => current + 1}
-            %{} -> Map.put(bag, char, 1)
+            %{^gc => current} -> %{bag | gc => current + 1}
+            %{} -> Map.put(bag, gc, 1)
           end
 
         string_to_bag(rest, bag, length + 1)
 
-      nil ->
+      [] ->
         {bag, length}
+
+      # Let's not count mismatched bytes in the bag
```

Comment:
This means the results of the new version are different than the previous one. For example, with previous Elixir versions:

```elixir
iex> String.bag_distance(<<?a, 200, 200>>, "ab")
0.33333333333333337
```

With the implementation of this PR:

```elixir
iex> String.bag_distance(<<?a, 200, 200>>, "ab")
0.6666666666666667
```

Is this intended? 🙂

Reply:
It was intentional but I could not reproduce the same behavior for jaro, so I will revert. Good catch! :)